Travis is broken 🤨

A couple of days ago we've re-opened https://github.com/TryGhost/Ghost/issues/7470, because we suffer from a test, which fails in Travis, but not locally.

It's always the same test:

```
  1) Spam Prevention API Ensure reset works: password grant type:
      AssertionError: expected 2 to equal 1
      + expected - actual
      -2
      +1
      
      at Assertion.fail (node_modules/should/cjs/should.js:275:17)
      at Assertion.value (node_modules/should/cjs/should.js:356:19)
      at core/test/functional/api/v0.1/spam_prevention_spec.js:95:44
```

[See example build.](https://travis-ci.org/TryGhost/Ghost/jobs/467055340)

**It looks like this is no longer random**, [any commit in master is red at the moment.](https://github.com/TryGhost/Ghost/commits/master)

**It looks like it's only failing on MySQL.**

Any help is appreciated 🤩👻

--- a/core/test/functional/api/v0.1/spam_prevention_spec.js
+++ b/core/test/functional/api/v0.1/spam_prevention_spec.js
@@ -87,6 +87,9 @@ describe('Spam Prevention API', function () {
     it('Ensure reset works: password grant type', function () {
             return executeRequests(userAllowedAttempts - 1, loginAttempt, owner.email, incorrectPassword)
             .then(() => loginAttempt(owner.email, correctPassword))
+            // CASE: login in with bad credentials twice - which would
+            // take us over the limit if the block hasn't been reset
+            .then(() => loginAttempt(owner.email, incorrectPassword))
             .then(() => loginAttempt(owner.email, incorrectPassword))
             .then((res) => {
                 // CASE: the reset means that we should be able to attempt to log in again

--- a/core/test/functional/api/v0.1/spam_prevention_spec.js
+++ b/core/test/functional/api/v0.1/spam_prevention_spec.js
@@ -87,13 +87,15 @@ describe('Spam Prevention API', function () {
     it('Ensure reset works: password grant type', function () {
             return executeRequests(userAllowedAttempts - 1, loginAttempt, owner.email, incorrectPassword)
             .then(() => loginAttempt(owner.email, correctPassword))
-            .then(() => {
-                return db.knex('brute').select()
-                    .then(function (rows) {
-                        // if reset works, the key is deleted and only one key remains in the database
-                        // the one key is the key for global block
-                        rows.length.should.eql(1);
-                    });
+            .then(() => loginAttempt(owner.email, incorrectPassword))
+            .then((res) => {
+                // CASE: the reset means that we should be able to attempt to log in again
+                // and not get a too many requests error
+                const error = res.body.errors[0];
+                should.exist(error.errorType);
+                res.statusCode.should.eql(422);
+                error.errorType.should.eql('ValidationError');
+                error.message.should.eql('Your password is incorrect.');
             });
     });
 });

